packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -381,6 +381,19 @@ abstract class SchedulerBinding extends BindingBase {
   Duration _epochStart = Duration.ZERO;
   Duration _lastRawTimeStamp = Duration.ZERO;
 
+  /// Prepares the scheduler for a non-monotonic change to how time stamps are calcuated.
+  ///
+  /// Callbacks received from the scheduler assume that their time stamps are
+  /// monotonically increasing. The raw time stamp passed to [handleBeginFrame]
+  /// is monotonic, but the scheduler might adjust those time stamps to provide
+  /// [timeDilation]. Without careful handling, these adjusts could cause time
+  /// to appear to run backwards.
+  ///
+  /// The [resetEpoch] function ensures that the time stamps are monotonic by
+  /// reseting the base time stamp used for future time stamp adjustments to the
+  /// current value. For example, if the [timeDilation] decreases, rather than
+  /// scaling down the [Duration] since the beginning of time, [resetEpoch] will
+  /// ensure that we only scale down the duration since [resetEpoch] was called.
```

Comment:
this comment is great. One additional thing it should mention is whether to call resetEpoch() before or after setting timeDilation (or if it doesn't matter).